### PR TITLE
Handle `@hosts` parameter on data_stream plugin

### PR DESCRIPTION
**Pull request by cosmo0920**
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

DataStream plugin should be able to handle `@hosts` parameter.

(check all that apply)
- [x] tests added
- [x] tests passing
- [x] README updated (if needed)
- [x] README Table of Contents updated (if needed)
- [x] History.md and `version` in gemspec are untouched
- [x] backward compatible
- [ ] feature works in `elasticsearch_dynamic` (not required but recommended)

### CODE REVIEW

**Comment by cosmo0920** [comment]
This issue is originally reported in https://github.com/fluent/fluent-plugin-opensearch/issues/5. But this issue also exists in this plugin.

File: test/plugin/test_out_elasticsearch_data_stream.rb [hunk 1]
```diff
@@ -437,6 +437,38 @@ def test_datastream_configure
...
+  def test_hosts_list_configure
+    config = %{
+      hosts            https://john:password@host1:443/elastic/,http://host2
+      path             /default_path
+      user             default_user
+      password         default_password
+      data_stream_name default
+    }
+    stub_elastic_info("https://host1:443/elastic//", "7.9.0",
+                         {'Authorization'=>'Basic am9objpwYXNzd29yZA=='})
```
Comment (kenhys) [inline test/plugin/test_out_elasticsearch_data_stream.rb:449]: `"Authorization" => "Basic #{Base64.encode64('john:password').split.first}"` or something may be readable.

Comment (cosmo0920) [reply test/plugin/test_out_elasticsearch_data_stream.rb]: Fixed in https://github.com/uken/fluent-plugin-elasticsearch/pull/942/commits/67390bcfa1ece1d42a56d3584ce14d18b902c561

**Comment by cosmo0920** [comment]
@kenhys Any comments?

**Review by kenhys — APPROVED** [review]
LGTM

**Comment by AurelPaulovic** [comment]
Hello guys, this still does not work - confirmed both on 5.1.5 and 5.2.0.

In https://github.com/fluent/fluent-plugin-opensearch/pull/21 you can see that there is a second part to the fix - actually using the `hosts` parameter.

**Pull request merged**